lib/atomic_cache/storage/dalli.rb

```diff
@@ -33,12 +33,14 @@ def add(key, new_value, ttl, user_options=nil)
 
       def read(key, user_options=nil)
         user_options ||= {}
-        @dalli_client.read(key, user_options)
+        raw = @dalli_client.read(key, user_options)
```

Comment:
What do you think about naming this variable `marshaled` ? It might be more clear.

On the other hand, it looks like the `marshal` method might not marshal the objects depending on the options. I wonder if you might change the structure to only call `marshal` if it should be done. That could make the code slightly more understandable.